framework/db/ActiveQuery.php

```diff
@@ -721,4 +724,35 @@ public function viaTable($tableName, $link, callable $callable = null)
 
         return $this;
     }
+
+    /**
+     * Define an alias for the table defined in [[modelClass]].
+     *
+     * This method will adjust [[from]] so that an already defined alias will be overwritten.
+     * If none was defined, [[from]] will be populated with the given alias.
+     *
+     * @param string $alias the table alias.
+     * @return $this the query object itself
+     * @since 2.0.7
+     */
+    public function alias($alias)
```

Comment:
note, that this method is added for ActiveQuery only. The use case is e.g. the following:

```php
// in Post AR class
public function getAuthor()
{
     return $this->hasOne(User::classname(), ['id', 'author_id'])->alias('author');
}
```